sql/src/main/antlr/OpenSearchSQLParser.g4

```diff
@@ -244,6 +244,7 @@ datetimeConstantLiteral
     : CURRENT_DATE
     | CURRENT_TIME
     | CURRENT_TIMESTAMP
+    | DAY_OF_MONTH
```

Comment:
Same as https://github.com/Bit-Quill/opensearch-project-sql/pull/195#discussion_r1052713305

Reply:
Fixed in 878a6a152955d3cbbc1a21c85ff2d900a6bc785c